snippet_ranger/model2/base_decompose.py

```diff
@@ -0,0 +1,69 @@
+from typing import Iterable
+
+from modelforge import Model
+from ast2vec import Model2Base
+
+
+class Model2BaseDecompose(Model2Base):
+    """
+    Class for model entry filtration, decomposition, filtration again and merging.
+    Produce exactly one model from one, but it has more entries because of decomposition.
+    MODEL_FROM should be iterable.
+    """
+    def __init__(self, *args, **kwargs):
```

Comment:
I think it is better to have two classes. In general, I need some code snippets and may be functions are not good. Later I can add others inheritors.